node/src/actors/chain_manager/mining.rs

```diff
@@ -893,27 +896,27 @@ fn build_block(
             }
         };
 
-        let new_block_weight = block_weight + transaction_weight;
-
-        if new_block_weight <= max_block_weight {
-            value_transfer_txns.push(vt_tx.clone());
+        let new_vt_weight = vt_weight + transaction_weight;
```

Comment:
Not sure if an overflow is possible here, but better handle it just in case.

```suggestion
        let new_vt_weight = vt_weight.checked_add(transaction_weight);
```